modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -314,13 +314,18 @@ cvtArg (const char *filename, int lineno, char *arg, iocshArgBuf *argBuf,
         break;
 
     case iocshArgPersistentString:
-        argBuf->sval = (char *) malloc(strlen(arg) + 1);
-        if (argBuf->sval == NULL) {
-            showError(filename, lineno, "Out of memory");
+        if (arg != NULL) {
+            argBuf->sval = (char *) malloc(strlen(arg) + 1);
+            if (argBuf->sval == NULL) {
+                showError(filename, lineno, "Out of memory");
+                return 0;
+            }
+            strcpy(argBuf->sval, arg);
+            break;
+        } else {
+            showError(filename, lineno, "Unable to copy as provided arg was NULL");
```

Comment:
I don't think that `!arg` is an error and rather should be passed through as `!argBuf->sval`.

Reply:
I've updated this now so `argBuf->sval = NULL` in the event a null argument is passed.

Should we also test for `argBuff == NULL`? Or is it implicit that `argBuf` cannot be NULL?

Freddie also mentioned that we could potentially use the epics strdup?